client/components/SearchForm.jsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import _ from 'lodash';
+
+export default class SearchForm extends React.Component {
+  search(e) {
+    const searchQ = this.input.value
+    if (!searchQ) {
+        return this.props.onItemsChange([]);
+    }
+
+    $.get(`api/search/?q=${searchQ}`).done((data) => {
+      const items = JSON.parse(data).items;
+      this.props.onItemsChange(items);
+    });
+  }
+
+  render() {
+    return (
+      <form action="" method="">
+        <input type="search"
+               placeholder="Search"
+               id="search"
+               onChange={_.debounce(this.search.bind(this), 300)}
```

Comment:
note: we should check if we have limit a of the youtube api requests and probably increase this value.